ruby/lib/minitest/queue.rb

```diff
@@ -207,12 +207,18 @@ def __run(*args)
 
     def run_from_queue(reporter, *)
       queue.poll do |example|
-        result = example.run
-        failed = !(result.passed? || result.skipped?)
 
         if example.flaky?
+          if queue.run_flakey_tests?
+            result = example.run
+          else
+            result = MiniTest::Skip.new
```

Comment:
`Skip` is an assertion error not a result. I'm not very familiar with minitest internals, but I believe you may have to do something like this here:

```ruby
example.failures << Minitest::Skip.new
result = Minitest::Result.from(example)
```

As `Skip` is supposed to be `raised`, you may need to manually set the backtrace. I've found [this code](https://github.com/ivantsepp/minitest-skip/blob/master/lib/minitest/skip_plugin.rb) that shows you how to do it.

It would be good to have a test case covering this scenario too.

Reply:
thanks, just wanted to check the approach before moving forward :)